src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/Target_ARM/ARMReadyToRunGenericHelperNode.cs

```diff
@@ -154,17 +149,11 @@ protected sealed override void EmitCode(NodeFactory factory, ref ARMEmitter enco
                             encoder.EmitLDR(encoder.TargetRegister.Arg0, encoder.TargetRegister.Arg0);
                             encoder.EmitJMP(factory.HelperEntrypoint(HelperEntrypoint.EnsureClassConstructorRunAndReturnGCStaticBase));
                         }
-                        */
```

Comment:
```
// If we get the required state it will be necessary to restore the original value before returning
encoder.EmitPUSH(encoder.TargetRegister.Arg0);
```

It would be better to use R2 here like in the `ReadyToRunHelperId.GetThreadStaticBase` case. PUSH/POP without matching unwind information is not good.